File: livingdoc-api/src/main/java/org/livingdoc/api/documents/FailFast.java
package org.livingdoc.api.documents;


import java.lang.annotation.*;

/**
 * This annotation is used to mark a document to use fail fast behaviour. Whenever a test fails, the whole execution
 * should stop.
 */
@Target(ElementType.TYPE)
@Retention(RetentionPolicy.RUNTIME)
public @interface FailFast {
    /**
     * The Array of Exceptions that should lead to a fast fail
     */
    Class<? extends Throwable>[] onExceptionTypes() default Throwable.class;

}

File: livingdoc-api/src/main/kotlin/org/livingdoc/api/documents/FailFast.kt
package org.livingdoc.api.documents

import kotlin.reflect.KClass

/**
 * This annotation is used to mark a document to use fail fast behaviour. Whenever a test fails, the whole execution
 * should stop.
 */
@Target(AnnotationTarget.CLASS)
@Retention(AnnotationRetention.RUNTIME)
annotation class FailFast(
    /**
     * The Array of Exceptions that should lead to a fast fail
     */
    val onExceptionTypes: Array<KClass<out Throwable>> = [Throwable::class]
)
